encfs/encfssh

```diff
@@ -28,42 +29,63 @@ if [ ! -z "$2" ]; then
 	    mount_public=true
 	fi
     done
-    [ -d "$unenc_dir" ] || mkdir $unenc_dir
+    [ -d "$unenc_dir" ] || mkdir -- "$unenc_dir"
 else
     unenc_dir=$(mktemp -d .XXXXXXXX)
 fi
 
 if [ ! -d "$enc_dir" ]; then
-    mkdir $enc_dir
+    mkdir -- "$enc_dir"
 fi
 
 enc_dir=$(canonicalize "$enc_dir")
 unenc_dir=$(canonicalize "$unenc_dir")
 
-options=""
-if $unenc_dir_given; then
-    if $mount_public; then
-	options="-- -o allow_other"
+# clear command-line argument list; hereafter used for options
+set --
+if [ "$unenc_dir_given" = "true" ]; then
+    if [ "$mount_public" = "true" ]; then
+	set -- -- -o allow_other
     fi
 fi
 
 # Attach the directory and change into it
-if encfs $enc_dir $unenc_dir $options; then :; else
+if encfs "$enc_dir" "$unenc_dir" "$@"; then :; else
     echo "encfs failed"
-    rmdir $unenc_dir
+    rmdir -- "$unenc_dir"
     exit 1
 fi
-if ! $unenc_dir_given; then
-    chmod 700 $unenc_dir
+if ! [ "$unenc_dir_given" = "true" ]; then
+    chmod 700 "$unenc_dir"
 fi
-echo "Directory is $unenc_dir"
-orig_dir=$(pwd)
-cd $unenc_dir
+echo "Directory is $unenc_dir" >&2
+cd "$unenc_dir" || exit
 
 # Fall back to umount if fusermount is not available (e.g., on OS X)
-FUSE_UMOUNT="$(which 2>/dev/null fusermount)"
+FUSE_UMOUNT="$(command -v fusermount)"
 FUSE_UMOUNT="${FUSE_UMOUNT:+fusermount -u}"
 FUSE_UMOUNT="${FUSE_UMOUNT:-umount}"
```

Comment:
Why? The chain of commands can't possibly fail -- the fallback is just assigning the `umount` command, after all.